File: spring-boot-s3-java-maven-aws-java-sdk/src/main/java/io/pivotal/tsalm/s3/infrastructure/config/S3ConfigurationProperties.java
package io.pivotal.tsalm.s3.infrastructure.config;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@ConfigurationProperties("s3properties")
@Component
class S3ConfigurationProperties {

    private String endpointUrl;
    private String region;
    private String accessKey;
    private String secretKey;

    public String getEndpointUrl() {
        return endpointUrl;
    }

    public void setEndpointUrl(String endpointUrl) {
        this.endpointUrl = endpointUrl;
    }

    public String getRegion() {
        return region;
    }

    public void setRegion(String region) {
        this.region = region;
    }

    public String getAccessKey() {
        return accessKey;
    }

    public void setAccessKey(String accessKey) {
        this.accessKey = accessKey;
    }

    public String getSecretKey() {
        return secretKey;
    }

    public void setSecretKey(String secretKey) {
        this.secretKey = secretKey;
    }
}

File: spring-boot-s3-kotlin-gradle-aws-java-sdk/src/main/kotlin/io/pivotal/tsalm/s3/infrastructure/config/S3ConfigurationProperties.kt
package io.pivotal.tsalm.s3.infrastructure.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@ConfigurationProperties("s3properties")
@Component
class S3ConfigurationProperties {

    lateinit var endpointUrl: String
    lateinit var region: String
    lateinit var accessKey: String
    lateinit var secretKey: String
}
